agent/agent.go

```diff
@@ -188,20 +196,99 @@ func (agent *Agent) Stop(ctx context.Context) {
 
 func (agent *Agent) healthCheck() {
 	//TODO: make this ticker configurable
-	timer := time.NewTicker(15 * time.Second)
+	timer := agent.clock.NewTicker(15 * time.Second)
 	for {
 		select {
 		case <-agent.ctx.Done():
-		case <-timer.C:
+		case <-timer.Chan():
 			report := agent.calculateHealth()
 			if report != nil {
 				agent.lastHealth = report
-				agent.opampClient.SetHealth(report)
+				if err := agent.opampClient.SetHealth(report); err != nil {
+					agent.logger.Errorf(context.Background(), "Could not report health to OpAMP server: %v", err)
+				}
 			}
+
+			traceUsage, ok := agent.metrics.Get("bytes_received_trace")
+			if !ok {
+				agent.logger.Errorf(context.Background(), "unexpected missing trace usage metric")
+			}
+			logUsage, ok := agent.metrics.Get("bytes_received_log")
+			if !ok {
+				agent.logger.Errorf(context.Background(), "unexpected missing log usage metric")
+			}
+
+			now := agent.clock.Now()
+			agent.usageTracker.Add(newTraceCumulativeUsage(traceUsage, now))
+			agent.usageTracker.Add(newLogCumulativeUsage(logUsage, now))
+		}
+	}
+}
+
+func (agent *Agent) usageReport() {
```

Comment:
Since this is specific to sending the report and not accumulating the metrics and we update the name to reflect that?